### PR TITLE
COMMERCE-12393

### PR DESCRIPTION
**Pull request by Brunno0liveira**
https://liferay.atlassian.net/browse/COMMERCE-12393

### CODE REVIEW

**Comment by Brunno0liveira** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 1]
```diff
@@ -440,6 +440,80 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12393. As a buyer I can view the UOMs sorted by priority."
```
Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:443]: @description = "COMMERCE-12393. As a buyer I can view the UOMs sorted by priority on the UOM selector from the product details page

Comment (22LucaGiannotta22) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:443]: @description = "COMMERCE-12393. As a buyer I can view the UOMs sorted by priority on the UOM selector from the product details page."

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 2]
```diff
@@ -440,6 +440,80 @@ definition {
...
+
+		task ("Then the buyer user can assert the UOMs are sorted by priority ") {
+			for (var indexAndUOMName : list "1:uom1,2:uom3,3:uom4") {
+				var index = StringUtil.extractFirst(${indexAndUOMName}, ":");
+				var uomName = StringUtil.extractLast(${indexAndUOMName}, ":");
+
+				AssertElementPresent(
+					index = ${index},
+					locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECTOR_VALUE_BY_INDEX",
+					uomName = ${uomName});
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:503]: - The highest priority UOM should be pre-selected(UOM2). I noticed that we are just checking with an AssertElementPresent and an Idex, but we don't verify the content of what is written. At that moment, you should be able to use the AssertSelectedLabel with this path here: CommerceFrontStore#FS_PRODUCT_UOM_SELECT

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase [hunk 3]
```diff
@@ -440,6 +440,80 @@ definition {
...
+			}
+
+			AssertElementNotPresent(
+				locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECTOR_VALUE",
+				uomName = "uom2");
+
+			FrontStore.assertProductDetails(
+				productListPrice = "25.00",
+				productName = "ABS Sensor",
+				productSKU = "MIN93015");
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase:513]: We are not checking the quantity in the product details

**Review by pietrobocale — CHANGES_REQUESTED** [review]
The review is done. Please take a look at the comments.

Thanks!

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
Review Passed

Thanks!

**Review by pietrobocale — APPROVED** [review]

**Pull request closed without merging**